packages/diagrams/frontend/sirius-components-diagrams/src/renderer/delete/useDiagramDelete.tsx

```diff
@@ -77,24 +77,39 @@ export const useDiagramDelete = (): UseDiagramDeleteValue => {
   }, [deleteElementsData, deleteElementsError]);
 
   const onDelete = useCallback((event: React.KeyboardEvent<Element>) => {
-    const { key } = event;
-    /*If a modifier key is hit alone, do nothing*/
+    const { key, target } = event;
     const isTextField = event.target instanceof HTMLInputElement || event.target instanceof HTMLTextAreaElement;
-    if ((event.altKey && key === 'Alt') || (event.shiftKey && key === 'Shift') || isTextField) {
+
+    /*If a modifier key is hit alone, do nothing*/
+    if ((event.altKey && key === 'Alt') || (event.shiftKey && key === 'Shift') || isTextField || readOnly) {
+      return;
+    }
+    /*If the palette is opened, do nothing*/
```

Comment:
1- Why?
2- This does not work, it only works half the time so why?

Why can't I click an edge and use the key `Delete` to delete it?

Reply:
If you open the palette you have the focus on the searchfield input and as such we want to keep the use of the `Delete `key to manipulate the text in the field.
You have to close the palette to Delete the element or click a tool.